diff -urN ppsspp-0.8/UI/NativeApp.cpp ppsspp-0.8-patched/UI/NativeApp.cpp
--- ppsspp-0.8/UI/NativeApp.cpp	2013-06-24 05:10:00.000000000 +1100
+++ ppsspp-0.8-patched/UI/NativeApp.cpp	2013-06-25 01:03:05.000000000 +1100
@@ -225,6 +225,7 @@
 	VFSRegister("", new DirectoryAssetReader("assets/"));
 #endif
 #endif
+	VFSRegister("", new DirectoryAssetReader("/usr/share/games/ppsspp/assets/"));
 	VFSRegister("", new DirectoryAssetReader(user_data_path.c_str()));
 
 	host = new NativeHost();
