diff -urN ppsspp-0.8-orig/Core/HW/atrac3plus.cpp ppsspp-0.8-patched/Core/HW/atrac3plus.cpp
--- ppsspp-0.8-orig/Core/HW/atrac3plus.cpp	2013-06-24 05:10:00.000000000 +1100
+++ ppsspp-0.8-patched/Core/HW/atrac3plus.cpp	2013-06-26 17:53:30.544177853 +1100
@@ -28,7 +28,7 @@
 namespace Atrac3plus_Decoder {
 
 	bool IsSupported() {
-#if (defined(_WIN32) && (defined(_M_IX86) || defined(_M_X64))) || defined(ARMEABI) || defined(ARMEABI_V7A) || defined(MACOSX)
+#if (defined(_WIN32) && (defined(_M_IX86) || defined(_M_X64))) || defined(ARMEABI) || defined(ARMEABI_V7A) || defined(MACOSX) || defined(__linux__)
 		return true;
 #else
 		return false;
@@ -59,6 +59,12 @@
 #endif
 #elif defined(__APPLE__)
 		return "libat3plusdecoder.dylib";
+#elif defined(__linux__)
+#ifdef __x86_64__
+		return "/usr/lib64/ppsspp/libat3plusdecoder.so";
+#else
+		return "/usr/lib/ppsspp/libat3plusdecoder.so";
+#endif
 #else
 		return "libat3plusdecoder.so";
 #endif
diff -urN ppsspp-0.8-orig/UI/MenuScreens.cpp ppsspp-0.8-patched/UI/MenuScreens.cpp
--- ppsspp-0.8-orig/UI/MenuScreens.cpp	2013-06-24 05:10:00.000000000 +1100
+++ ppsspp-0.8-patched/UI/MenuScreens.cpp	2013-06-26 18:02:33.000000000 +1100
@@ -738,10 +738,12 @@
 			UICheckBox(GEN_ID, x + 60, y += stride, a->T("Enable Atrac3+"), ALIGN_TOPLEFT, &g_Config.bEnableAtrac3plus);
 		}
 
+#if !defined __linux__
 		VLinear vlinear(30, 200, 20);
 		if (UIButton(GEN_ID, vlinear, 400, 0, a->T("Download Atrac3+ plugin"), ALIGN_LEFT)) {
 			screenManager()->push(new PluginScreen());
 		}
+#endif
 	}
 
 	UIEnd();
